textdb/textdb-storage/src/main/java/edu/uci/ics/textdb/storage/relation/RelationManager.java

```diff
@@ -0,0 +1,363 @@
+    public String getTableAnalyzer(String tableName) throws StorageException {
+        DataStore collectionCatalogStore = new DataStore(CatalogConstants.COLLECTION_CATALOG_DIRECTORY,
+                CatalogConstants.COLLECTION_CATALOG_SCHEMA);
+        DataReaderPredicate scanPredicate = new DataReaderPredicate(new MatchAllDocsQuery(), collectionCatalogStore);
+        scanPredicate.setIsPayloadAdded(false);
+        DataReader collectionCatalogReader = new DataReader(scanPredicate);
+
+        collectionCatalogReader.open();
+        ITuple nextTuple;
+        while ((nextTuple = collectionCatalogReader.getNextTuple()) != null) {
+            IField tableNameField = nextTuple.getField(CatalogConstants.COLLECTION_NAME);
+            // field must not be null
+            if (tableNameField == null) {
+                continue;
+            }
+            // table name must be the same (case insensitive)
+            String tableNameString = tableNameField.getValue().toString();
+            if (! tableNameString.toLowerCase().equals(tableName.toLowerCase())) {
+                continue;
+            }
+            IField analyzerField = nextTuple.getField(CatalogConstants.COLLECTION_LUCENE_ANALYZER);
+            if (analyzerField == null) {
+                break;
+            }
+            return analyzerField.getValue().toString();
+        }
+        collectionCatalogReader.close();
+
+        throw new StorageException(String.format("The analyzer for table %s is not found.", tableName));
+    }
+    
+    private static boolean checkCatalogExistence() {
+        return DataReader.checkIndexExistence(CatalogConstants.COLLECTION_CATALOG_DIRECTORY)
+                && DataReader.checkIndexExistence(CatalogConstants.SCHEMA_CATALOG_DIRECTORY);
+    }
+    
+    private void initializeCatalog() throws StorageException {
+        createTable(CatalogConstants.COLLECTION_CATALOG, 
+                CatalogConstants.COLLECTION_CATALOG_DIRECTORY, 
+                CatalogConstants.COLLECTION_CATALOG_SCHEMA,
+                LuceneAnalyzerConstants.standardAnalyzerString());
+        createTable(CatalogConstants.SCHEMA_CATALOG,
+                CatalogConstants.SCHEMA_CATALOG_DIRECTORY,
+                CatalogConstants.SCHEMA_CATALOG_SCHEMA,
+                LuceneAnalyzerConstants.standardAnalyzerString());
+    }
+    
+    
+    /**
+     * This function converts a attributeTypeString to FieldType (case insensitive). 
+     * It returns null if string is not a valid type.
+     * 
+     * @param attributeTypeStr
+     * @return FieldType, null if attributeTypeStr is not a valid type.
+     */
+    private static FieldType convertAttributeType(String attributeTypeStr) {
+        return Stream.of(FieldType.values())
+                .filter(typeStr -> typeStr.toString().toLowerCase().equals(attributeTypeStr.toLowerCase()))
+                .findAny().orElse(null);
+    }
+    
+    private static ITuple getTupleWithID(ITuple tuple, IDField _id) {
```

Comment:
Add a comment to explain the fact that we will add an `_id` field if it doesn't exist.